src/store/index.js

```diff
@@ -151,6 +185,15 @@ export default new Vuex.Store({
         .map(r => (reportingPeriod.id === r.id ? reportingPeriod : r))
         .sortBy('start_date')
         .value()
+    },
+    updateAllUploads (state, updatedUploads) {
+      state.allUploads = updatedUploads
+    },
+    addAlert (state, alert) {
+      Vue.set(state.alerts, randomId(), alert)
```

Comment:
i think `set` and `delete` below are vue-2 specific functions. in vue-3, i would declare the `alerts` object as `reactive` and then this would not be necessary.